src/lib/iptables.ts

```diff
@@ -0,0 +1,38 @@
+import * as Promise from 'bluebird';
+import * as _childProcess from 'child_process';
+
+const childProcess: any = Promise.promisifyAll(_childProcess);
+
+function clearAndAppendIptablesRule(rule: string): Promise<{}> {
+	return childProcess.execAsync(`iptables -D ${rule}`)
+		.catchReturn({})
+		.then(() => {
+			return childProcess.execAsync(`iptables -A ${rule}`);
+		});
+}
+
+function clearAndInsertIptablesRule(rule: string): Promise<{}> {
+	return childProcess.execAsync(`iptables -D ${rule}`)
+		.catchReturn({})
+		.then(() => {
+			return childProcess.execAsync(`iptables -I ${rule}`);
+		});
+}
+
+export function rejectOnAllInterfacesExcept(
+	allowedInterfaces: string[],
+	port: number,
+): Promise<{}> {
+	// We delete each rule and create it again to ensure ordering (all ACCEPTs before the REJECT/DROP).
+	// This is especially important after a supervisor update.
+	return Promise.each(allowedInterfaces, (iface) => {
+		return clearAndInsertIptablesRule(`INPUT -p tcp --dport ${port} -i ${iface} -j ACCEPT`);
+	})
+		.then(() => {
+			return clearAndAppendIptablesRule(`INPUT -p tcp --dport ${port} -j REJECT`)
```

Comment:
I think dropping the braces + return would be really helpful here to get rid of this super wonky indentation